app/services/invite_schools.rb

```diff
@@ -436,4 +448,14 @@ def send_section_41_invite_email(nomination_email)
   def logger
     @logger ||= Rails.logger
   end
+
+  def diy_school_cohorts_with_pending_partnerships
+    SchoolCohort.where(cohort_id: Cohort.current.id, induction_programme_choice: "design_our_own").joins(school: :partnerships).where(school: { partnerships: { challenge_reason: nil } })
+  end
+
+  def diy_school_cohorts_without_pending_partnerships
+    SchoolCohort
+      .where(cohort_id: Cohort.current.id, induction_programme_choice: "design_our_own")
+      .where.not(id: diy_school_cohorts_with_pending_partnerships)
+  end
```

Comment:
Should these be scopes on SchoolCohort?

Reply:
Some of them might be scopes already, and this is likely the only time we will ever use the others